README.md

```diff
@@ -1,17 +1,58 @@
 Coclobas
-========
+======
 
-Configurable Cloudy Batch Scheduler
+Coclobas provides an OCaml interface on top of Kubernetes, making it easy to programatically interface with cluster from the command-line and from code. It provides logging facilities, maintaining logs long after Kubernetes disposes of them. Finally, it makes it easy to submit arbitrary script to be run in any Docker container, which makes it easier than using raw Kubernetes to submit arbitrarily complex jobs.
+
+Right now Coclobas is Kubernetes focused and works with Google Container Engine, but we aim to support arbitrary Kubernetes deployments, as well as alternate scheduler types.
 
 
 Build
 -----
 
-Uses the development version of Solvuu-build:
+Coclobas uses the development version of Solvuu-build:
 
     opam pin add solvuu-build https://github.com/solvuu/solvuu-build.git
 
-Then,
+Then, clode this repo and:
 
     make
+    
+After pinning `solvuu-build`, you can also just `opam pin add coclobas https://github.com/hammerlab/coclobas` to get going quickly.
+
+Requirements
+-----
+
+`kubectl` must be installed local to the Coclobas cluster, as it is shelled out to in order to query the Kubernetes cluster being used to run the submitted job.
+
+
+Using Coclobas
+-----
+
+You will need to briefly configure Coclobas prior to using it. You also will
+need to set up a Google Container Engine (GCE) cluster and note its name. This
+is mostly a matter of choosing a directory (`--root`) for the Coclobas database
+to live in, and telling it where it can find your Kubernetes cluster
+(`--cluster-name`) on the GCE and which compute zone that cluster is in
+(`--cluser-zone`). Finally, you'll want to limit the total number of pods
+Coclobas can schedule (`--max-nodes`).
+
+```shell
+coclobas configure --root _configure_root/ \
+    --cluster-name some-cluster \
+    --cluster-zone us-east1-c \ # Zone 
+    --max-nodes 20  # The total number of pods Coclobas will schedule
```

Comment:
it's the maximal size the cluster will be allowed to have.
Coclobas will schedule more jobs, but they will be queued.